src/Avalonia.Animation/Animatable.cs

```diff
@@ -36,6 +36,9 @@ public IClock Clock
 
         private Transitions _transitions;
 
+        private Dictionary<AvaloniaProperty, IDisposable> _previousTransitions
```

Comment:
Perhaps don't allocate this collection until it's needed? Many controls won't have transitions and it'll save quite a bit of memory.

Reply:
Also will this prevent transitions from being GC'd where previously they would have been?

Reply:
Yep, i forgot to include the local diff for addressing that